lib/features/modeling/ChoreoModeling.js

```diff
@@ -108,4 +110,16 @@ ChoreoModeling.prototype.changeParticipant = function(bandShape, participant) {
     bandShape: bandShape,
     newParticipant: participant
   });
+};
+
+ChoreoModeling.prototype.setInitiatingBand = function(element) {
+  const initParticipantBand = element.parent.children.find(
+    x => x.businessObject.id === element.parent.businessObject.initiatingParticipantRef.id);
+  this._commandStack.execute('band.makeInitiating', {
+    choreography: element.parent,
+    nonInitBand: element,
+    initBand: initParticipantBand,
+    nonInitBandKind: element.diBand.participantBandKind,
+    initBandKind: initParticipantBand.diBand.participantBandKind
```

Comment:
Adding the band kinds to the context would probably fit better in the `preExecute` function of the handler.